### PR TITLE
WIP: Use 1.24 Function syntax

### PR DESCRIPTION
**Pull request by kevmoo**
Blocked on a new analyzer release - https://github.com/dart-lang/sdk/issues/30239

Using alpha pkg/analyzer to see if travis gets happy
Looks like we need to update to use the 1.25.0-dev SDK, too!

### CODE REVIEW

File: angular/lib/src/core/linker/app_view.dart [hunk 1]
```diff
@@ -584,7 +583,7 @@ abstract class AppView<T> {
     domRootRendererIsDirty = true;
   }
 
-  VoidFunc1<E> eventHandler0<E>(VoidFunc0 handler) {
+  void Function(E) eventHandler0<E>(void Function() handler) {
```
Comment (leonsenft) [inline angular/lib/src/core/linker/app_view.dart]: Are we decided on what syntax to use when declaring function parameters? Both are valid:
```dart
void Function(E) eventHandler0<E>(void Function() handler) {}

void Function(E) eventHandler0<E>(void handler()) {}
```
I'm in favor of the latter as it's more concise, but I can understand the desire to be consistent with the former.

Comment (matanlurey) [inline angular/lib/src/core/linker/app_view.dart]: I think we decided always the former, the latter may be removed in a future version of Dart.

Comment (kevmoo) [inline angular/lib/src/core/linker/app_view.dart]: And encouraging the new syntax is goodness.

```dart
// No error, but this handle has no argument type
foo(void handler(int)) {};

// Does what you expect
foo(void Function(int)) {};
```

**Comment by leonsenft** [comment]
LTGM, although some of the files still contain function typdefs. It won't let me comment on them directly but see `*_control_acccesors.dart` for `void registerOnChange(ChangeFunction fn)`.

**Comment by kevmoo** [comment]
> LTGM, although some of the files still contain function typdefs. It won't let me comment on them directly but see *_control_acccesors.dart for void registerOnChange(ChangeFunction fn).

Yeah, those are used across many files – left them for now until we decide what to do.

File: angular/lib/src/core/linker/app_view.dart (hunk 1 above)
Comment (leonsenft) [inline angular/lib/src/core/linker/app_view.dart]: In that case there are some old style function parameters in these files you could update as well.

Comment (kevmoo) [inline angular/lib/src/core/linker/app_view.dart]: Yeah...but that's hard to regexp

I'd almost want a lint for that

**Pull request closed without merging**